specs/AppLifecycle/Instancing/CustomEventArgs.md

```diff
@@ -0,0 +1,170 @@
+Custom Activated Event Args for App Redirection.
+===
+
+# Background
+
+Each AppInstance can redirect its activation to another AppInstance and pass in an [AppActivationArguments](https://docs.microsoft.com/windows/windows-app-sdk/api/winrt/microsoft.windows.applifecycle.appactivationarguments)
+object obtained [GetActivatedEventArgs](https://docs.microsoft.com/windows/windows-app-sdk/api/winrt/microsoft.windows.applifecycle.appinstance.getactivatedeventargs).  This works if all you want to do is pass the arguments to
+another instance.  However, the arguments can't be changed.  This is because
```

Comment:
So there's a few options for this:

* Send unmodified - what we have today
* Change current arguments value - if it's a Foo+FooArguments, modify the content of FooArguments but retain the "it's a Foo" kind
* Change arguments entirely - if it's a Foo+FooArguments activation, change the activation to a Bar+BarArguments
* Send contextual information - retain the Foo-ness of the activation, but this instance wants to send along some hints to the target instance, tagging this activation with more data

I think all of these are interesting. In the "change" modes, this PR is a good proposal. In the "send contextual information" one approach would be to add a ValueSet get/set property on the Arguments object; it starts empty, but can be populated with data and then that set is sent along to the target of the activation.